src/coreclr/jit/simd.cpp

```diff
@@ -290,143 +292,103 @@ CorInfoType Compiler::getBaseJitTypeAndSizeOfSIMDType(CORINFO_CLASS_HANDLE typeH
         // slow path search
         if (simdBaseJitType == CORINFO_TYPE_UNDEF)
         {
-            // Doesn't match with any of the cached type handles.
-            // Obtain base type by parsing fully qualified class name.
-            //
-            // TODO-Throughput: implement product shipping solution to query base type.
-            WCHAR  className[256] = {0};
-            WCHAR* pbuf           = &className[0];
-            int    len            = ArrLen(className);
-            int    outlen = info.compCompHnd->appendClassName((char16_t**)&pbuf, &len, typeHnd, true, false, false);
-            noway_assert(outlen >= 0);
-            noway_assert((size_t)(outlen + 1) <= ArrLen(className));
-            JITDUMP("SIMD Candidate Type %S\n", className);
-
-            if (wcsncmp(className, W("System.Numerics."), 16) == 0)
+            JITDUMP("SIMD Candidate Type %s\n", className);
+
+            if (strcmp(className, "Vector`1") == 0)
             {
-                if (wcsncmp(&(className[16]), W("Vector`1["), 9) == 0)
-                {
-                    size = getSIMDVectorRegisterByteLength();
+                CORINFO_CLASS_HANDLE typeArgHnd = info.compCompHnd->getTypeInstantiationArgument(typeHnd, 0);
+                simdBaseJitType                 = info.compCompHnd->asCorInfoType(typeArgHnd);
```

Comment:
> `asCorInfoType`

Will the case of `Vector<MyEnum>` be rejected properly?

Reply:
I probably need `getTypeForPrimitiveNumericClass` instead.

Reply:
In fact, I can see that the code below for hw intrinsics is doing exactly this but with `getTypeForPrimitiveNumericClass`, so looks like that's definitely the way to go.